src/util/Conversions.h

```diff
@@ -272,7 +272,7 @@ string convertFloatToIndexWord(const string& orig, size_t nofExponentDigits,
 }
 
 // _____________________________________________________________________________
-string convertIndexWordToFloat(const string& indexWord) {
+string convertIndexWordToFloatString(const string& indexWord) {
```

Comment:
These two functions still share a lot of code, maybe this one should really just call `convertIndexWordToFloatValue`?

Reply:
I replaced the convertIndexWordToFloatString implementation with a call to `std::to_string(convertIndexWordToFloat(indexWord));`

Reply:
While testing the new implementation I ran into the problem, that the conversion using the float value suffers from small precision errors, which currently breaks the unit tests. This could be especially problematic with integers, which are internally stored as floats, as it might not be possible to represent the integer exactly.

Reply:
How, where and why do we store integers in floats? I'm a little confused, I thought this part is for stuff marked `^^xsd:float` or similar?

Reply:
https://github.com/ad-freiburg/QLever/blob/ee89fdc51b6eef788265e6524c2d550aaa96cdd2/src/util/Conversions.h#L114-L122

Reply:
Hmm, I don't know, I really don't like this part. But it seems like a separate issue, can you maybe create a GitHub issue and for now keep the code separate with a TODO and a pointer to why is hasn't been done yet. I really think this shouldn't block GROUPING and we should go ahead with merging this PR.